### DISCUSSION TITLE
non interactive mode for `changeset add`?

### DISCUSSION THREAD

**Discussion by pigoz** [changesets/changesets#1118] [Q&A] [answered]
Hi, is it possible to run changeset in non interactive mode? Most of the times I want to add a changeset from my last commit
i.e.: `changeset add --minor --message "$(git log -1 --pretty=%B)"`

**Comment by Andarist** [comment]
Currently, there is no way to do that. We could consider accepting a PR that would introduce this. To make it work you would probably also have to introduce smth like `--accept-changed` to automatically "select" changed packages.

**Reply by Sh031224** [reply]
I opened a pull request with an implementation similar to the behavior the questioner said.  Please check this. #1121

**Reply by Nelvia-07** [reply]
need it urgently, pls merge it

**Reply by JGJP** [reply]
@Andarist please have a look at the PR and merge, if you need help maintaining I'll offer my services

**Comment by Shurtu-gal** [comment]
@Andarist the PR has been created a long ago. Is there anything on that?

**Comment by Netail** [comment, marked as answer]
Released in v3 I believe, with https://github.com/changesets/changesets/pull/1121
